source/opt/passes.h

```diff
@@ -74,6 +79,132 @@ class EliminateDeadConstantPass : public Pass {
   bool Process(ir::Module*) override;
 };
 
+// The optimization pass for folding SpecConstantOp and SpecConstantComposite
+// instructions. After freezing, spec constants defined with Spec ID are
+// specialized into front-end constants. This pass folds the SpecConstantOp
+// instructions and SpecConstantComposite instructions that depend on those
+// frozen spec constants.
+class FoldSpecConstantOpAndCompositePass : public Pass {
+ public:
+   // Data structure to record the in-operand of front-end constants.
+   // ScalarValue stores the value of of scalar constants. CompositeValue
+   // stores a vector of its component ids. Value is a abstract class of
+   // ScalarValue and CompositeValue.
+  struct ScalarValue;
+  struct CompositeValue;
+  struct Value {
```

Comment:
There is already inheritance involved. I think we should use `class` here as per the C++ style guilde: https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes.